tests/ui/abi/custom.rs

```diff
@@ -0,0 +1,74 @@
+// Test that `extern "custom"` functions can be called from assembly, and defined using a naked
+// function, and `global_asm!` with an `extern "custom"` block.
+//
+//@ run-pass
+//@ only-x86_64
+#![feature(abi_custom)]
+
+use std::arch::{asm, global_asm, naked_asm};
+
+#[unsafe(naked)]
+unsafe extern "custom" fn double(a: u64) -> u64 {
+    naked_asm!("add rax, rax", "ret");
+}
```

Comment:
For now, I think I'd suggest just not allowing parameters and a return type -- or putting this under a separate feature gate.

We could always allow this later.  It seems like one of the few question marks here, so it'd be better to do the minimal thing to start.